components/external-link/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { compact, uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+
+/**
+ * Internal dependencies
+ */
+import Dashicon from '../dashicon';
+import './style.scss';
+
+function ExternalLink( { href, children, className, rel = '', ...additionalProps } ) {
+	rel = uniq( compact( [
+		...rel.split( ' ' ),
+		'external',
+		'noreferrer',
+		'noopener',
+	] ) ).join( ' ' );
+	const classes = classnames( 'new-window-icon', className );
+	return (
+		<a { ...additionalProps } className={ classes } href={ href } target="_blank" rel={ rel }>
+			{ children }
+			<span className="screen-reader-text">{ __( '(opens in a new window)' ) }</span>
```

Comment:
Minor: Elsewhere in the WordPress codebase, this string is preceded with a translator comment `translators: accessibility text`. My main concern would be if omitting the translator comment would create a separate string entry in GlotPress (I'm not sure)

https://translate.wordpress.org/projects/wp/dev/fr/default?filters%5Bterm%5D=opens+in+a+new+window&filters%5Buser_login%5D=&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=desc

Otherwise this could be:

```jsx
<span className="screen-reader-text">
	{
		/* translators: accessibility text */
		__( '(opens in a new window)' )
	}
</span>
```